### PR TITLE
Add extension request number to extension card

### PR DESCRIPTION
**Pull request by MayurLund**
Displays the Extension Request Number to the SU on the dashboard site.

Since previously created Extension requests does not contain requestNumber field so I am displaying it by default as 1.

**Before Changes :** Extension Request Number detail is not present
image

**After Changes :** It Displays the Extension Request Number as well as handles the case where previous extension request number field is absent so it displays 1 for that.
image

**Added the test for it too :**

image

### CODE REVIEW

**Comment by Aryex82** [comment]
Please add tests

**Comment by MayurLund** [comment]
> Please add tests

Hi @Ajeyakrishna-k , I have added test case for it. Could you please review it?

**Review by kotesh-arya — APPROVED** [review]

**Review by ardourApeX — APPROVED** [review]
LGTM

File: __tests__/extension-requests/extension-requests.test.js [hunk 1]
```diff
@@ -563,4 +563,32 @@ describe('Tests the Extension Requests Screen', () => {
...
+    title = await page.$('.header h1');
+    searchBar = await page.$('#search');
+    filterButton = await page.$('#filter-button');
+    extensionCardsList = await page.$$('.extension-card');
+    extensionRequestsElement = await page.$('.extension-requests');
+
+    expect(title).toBeTruthy();
+    expect(searchBar).toBeTruthy();
+    expect(filterButton).toBeTruthy();
+    expect(extensionCardsList.length).toBe(4);
```
Comment (Ajeyakrishna-k) [inline __tests__/extension-requests/extension-requests.test.js:580]: why are we verifying all the elements in the page when the test is to check only the request number and value?

File: extension-requests/script.js [hunk 2]
```diff
@@ -372,141 +340,143 @@ async function createExtensionCard(data) {
...
+    datesContainer.appendChild(extensionRequestNumberContainer);
+
+    const extensionRequestNumber = createElement({
+      type: 'span',
+      attributes: { class: 'card-row-text extension-request-number' },
+      innerText: 'Request No.:  ',
+    });
+    extensionRequestNumberContainer.appendChild(extensionRequestNumber);
+
+    //Since previous extensions does not contain requestNumber field
```
Comment (Ajeyakrishna-k) [inline extension-requests/script.js:443]: Can we add a test case for this case as well, i.e., when the request number is not returned in the API response

File: extension-requests/script.js [hunk 3]
```diff
@@ -372,141 +340,143 @@ async function createExtensionCard(data) {
...
+    const requestNumber =
+      data.requestNumber !== undefined && data.requestNumber !== null
+        ? data.requestNumber
+        : '1';
```
Comment (Ajeyakrishna-k) [inline extension-requests/script.js:447]: Can we simplify this : 
eg: 
```javascript
const requestNumber = data.requestNumber || '1';
```

File: __tests__/extension-requests/extension-requests.test.js (hunk 1 above)
Comment (MayurLund) [reply __tests__/extension-requests/extension-requests.test.js]: Removed

File: extension-requests/script.js (hunk 3 above)
Comment (MayurLund) [reply extension-requests/script.js]: Thanks. I have made the necessary change.

File: __tests__/extension-requests/extension-requests.test.js [hunk 4]
```diff
@@ -586,4 +586,28 @@ describe('Tests the Extension Requests Screen', () => {
...
+    );
+    expect(cardNumber1Value).toBe('5');
+
+    extensionRequestNumberText = extensionRequestNumberContainer[2];
+    expect(extensionRequestNumberText).toBeTruthy();
+
+    const cardNumber2Value = await extensionRequestNumberContainer[3].evaluate(
+      (node) => node.textContent,
+    );
+    expect(cardNumber2Value).toBe('1');
```
Comment (Ajeyakrishna-k) [inline __tests__/extension-requests/extension-requests.test.js:611]: - Each test should have single responsibility. 

- If this test fails a person without any context will just know that there is some issue with the extension request number but will not know the case for which it failed.

- To fix, please move this case(check for when there is no value from the api response) to a different test and update the test name to specify the exact case you are testing.

**Comment by Ajeyakrishna-k** [comment]
Please update the screenshots, the design for task details is different in before and after.

Before: 
Screenshot 2023-10-02 at 3 58 58 PM
After: 
Screenshot 2023-10-02 at 3 59 04 PM

**Review by Achintya-Chatterjee — COMMENTED** [review]
@MayurLund can you please add a working video ?

**Comment by MayurLund** [comment]
> @MayurLund can you please add a working video ?

Hi @Achintya-Chatterjee Just Request No field change is there . I have added before and after changes and highlighted it too.

**Comment by MayurLund** [comment]
> Please update the screenshots, the design for task details is different in before and after.
> 
> Before: Screenshot 2023-10-02 at 3 58 58 PM After: Screenshot 2023-10-02 at 3 59 04 PM

Hi @Ajeyakrishna-k ,Thanks for bringing this in notice. I have updated now!

File: __tests__/extension-requests/extension-requests.test.js (hunk 4 above)
Comment (MayurLund) [reply __tests__/extension-requests/extension-requests.test.js:611]: Thanks for pointing that out! I have made the necessary changes and ensured each test has a single responsibility. Appreciate your input!

**Review by Ajeyakrishna-k — APPROVED** [review]

**Review by Achintya-Chatterjee — COMMENTED** [review]
@MayurLund please get it reviewed by one member

**Review by kotesh-arya — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**